allennlp/training/trainer.py

```diff
@@ -343,21 +353,20 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
             train_generator_tqdm = train_generator
 
         cumulative_batch_size = 0
-        for batch in train_generator_tqdm:
+        for batches in lazy_groups_of(train_generator_tqdm, self._num_gradient_accumulation_steps):
```

Comment:
Nit: Maybe `accumulation_groups` (or `batch_group` or whatever) instead of `batches`? Might be confusing to a reader see lines like `cur_batch = sum(training_util.get_batch_size(batch) for batch in batches)`.